keras_cv/layers/preprocessing/random_resized_crop.py

```diff
@@ -0,0 +1,140 @@
+             Defaults to `"reflect"`.
+        fill_value: fill_value in the `ImageProjectiveTransformV3` op.
+             A `Tensor` of type `float32`. The value to be filled when fill_mode is
+             constant".  Defaults to `0.0`.
+        seed: Integer. Used to create a random seed.
+    """
+    def __init__(self,
+                 target_size,
+                 aspect_ratio_factor=(3. / 4., 4. / 3.),
+                 area_factor=(0.08, 1.0),
+                 interpolation="bilinear",
+                 fill_mode="reflect",
+                 fill_value=0.0,
+                 seed=None,
+                 **kwargs):
+        super(RandomResizedCrop, self).__init__(seed=seed, **kwargs)
+
+        self.target_size = target_size
+        self.aspect_ratio_factor = aspect_ratio_factor
+        self.area_factor = preprocessing.parse_factor(area_factor,
+                                                      param_name="area_factor",
+                                                      seed=seed)
+
+        self.interpolation = interpolation
+        self.fill_mode = fill_mode
+        self.fill_value = fill_value
+        self.seed = seed
+
+        if area_factor == 0.0 and aspect_ratio_factor == 0.0:
+            warnings.warn(
+                "RandomResizedCrop received both `area_factor=0.0` and "
+                "`aspect_ratio_factor=0.0`. As a result, the layer will perform no "
+                "augmentation.")
+
+    def get_random_transformation(self,
+                                  image=None,
+                                  label=None,
+                                  bounding_box=None):
+        area_factor = self.area_factor()
+        aspect_ratio = tf.random.uniform((),
+                                         minval=self.aspect_ratio_factor[0],
+                                         maxval=self.aspect_ratio_factor[1],
+                                         dtype=tf.float32)
+
+        new_height = tf.clip_by_value(
+            tf.sqrt(area_factor / aspect_ratio), 0.0,
+            1.0)  # to avoid unwanted/unintuitive effects
+        new_width = tf.clip_by_value(tf.sqrt(area_factor * aspect_ratio), 0.0,
+                                     1.0)
+
+        height_offset = tf.random.uniform(
+            (),
+            minval=tf.minimum(0.0, 1.0 - new_height),
+            maxval=tf.maximum(0.0, 1.0 - new_height),
+            dtype=tf.float32,
+        )
+
+        width_offset = tf.random.uniform(
+            (),
+            minval=tf.minimum(0.0, 1.0 - new_width),
+            maxval=tf.maximum(0.0, 1.0 - new_width),
+            dtype=tf.float32,
+        )
+
+        y1 = height_offset
+        y2 = height_offset + new_height
+        x1 = width_offset
+        x2 = width_offset + new_width
+
+        return [[y1, x1, y2, x2]]
+
+    def augment_image(self, image, transformation):
+        image = tf.expand_dims(image, axis=0)
```

Comment:
Does `BaseImageAugmentationLayer` automatically make this a vectorized function? Otherwise, should we add a check to see if `image` already has a leading batch dimension? Happy to review other resources to make my understanding of the architecture more refined.